src/filters/taxonomy.js

```diff
@@ -0,0 +1,107 @@
+'use strict';
+
+const fs = require('fs');
+
+const yaml = require('js-yaml');
+const _ = require('lodash');
+
+const taxonomy = yaml.safeLoad(
+  // eslint-disable-next-line no-sync
+  fs.readFileSync('./content/_data/taxonomy.yaml', 'utf8'),
+);
+
+/* @docs
+label: fromTaxonomy
+category: general
+note: |
+  Access the taxonomy directly.
+  This can also be done in nunjucks,
+  using the `taxonomy` global variable,
+  but a filter makes that available
+  from inside macros,
+  without passing it in as an argument.
+params:
+  type:
+    type: oss | post
+    note: Any taxonomy defined in the `taxonomy.yaml` data file
+  find:
+    type: any
+    note: Find an object in the taxonomy, using lodash `_.find()`
+  get:
+    type: string
+    default: undefined
+    note: Optionally return a single attribute of the found object
+*/
+const fromTaxonomy = (type, find, get) => {
+  const found = _.find(taxonomy[type], find);
+  return found && get ? found[get] : found;
+};
+
+/* @docs
+label: ossGroups
+category: open-source
+note: |
+  Group a collection of open-source project-pages
+  based on the type of OSS contribution made.
+  Groups are defined in the `taxonomy.yaml` data file.
+params:
+  collection:
+    type: array
+    note: containing 11ty page objects
+  only:
+    type: string
+    default: undefined
+    note: Optionally return a single group, based on the group title
+*/
+const ossGroups = (collection, only) => {
+  const grouped = {};
+
+  taxonomy.oss.forEach((type) => {
+    const pages = collection.filter((page) =>
+      type.show.includes(page.data.oss),
+    );
+    if (pages.length) {
+      grouped[type.title] = pages;
+    }
+  });
```

Comment:
I wonder if we want to be able to sort these manually (either here or in taxonomy or oss.macros)? I'm not sure "date started" is really what we want for the OSS page?

Reply:
Sure. We're not sorting at all in this code, are we? We're just grouping. Feel free to propose how you think they should sort.

Reply:
Nah, I think we're relying on the default sort? I guess maybe rather than sort I'm thinking of "featured".

Reply:
Added to backlog: https://trello.com/c/dGgqSrQ9/58-i-want-featured-oss-projects